### PR TITLE
feat: cli tool for filtering swagger docs

### PR DESCRIPTION
**Pull request by jcmcnamee**
## Changes

- Adds `swagfilter` as a top-level subcommand: `jx3-openapi-generation swagfilter --strip-tag=<tag> [--input=<path>] [--output=<path>]`
- Core logic in `pkg/swagfilter/` parses a Swagger 2.0 spec via `github.com/go-openapi/spec`, strips all occurrences of the named tag from every operation's `tags` array, and writes the result back out
- Thin Cobra command in `pkg/cmd/swagfilter/` handles file I/O and flag parsing; `--output` defaults to overwriting `--input` in place
- 9 unit tests covering tag stripping, preservation of non-path fields, error cases, and edge cases

## Context

Pipelines often need to remove internal-only tags (e.g. `external`) from a Swagger spec before passing it to the generator — for example to exclude operations from a public-facing client. This command provides that capability as part of the existing toolchain, avoiding the need for a separate binary or bespoke shell scripting.

### CODE REVIEW

File: pkg/swagfilter/filter_test.go [hunk 1]
```diff
@@ -0,0 +1,218 @@
...
+
+func TestStripTagFromSpec_MaestroFixture(t *testing.T) {
+	data, err := os.ReadFile("testdata/maestro.json")
+	require.NoError(t, err)
+
+	result, err := swagfilter.StripTagFromSpec(data, "external")
+	require.NoError(t, err)
+
+	require.NoError(t, os.WriteFile("testdata/maestro_filtered.json", result, 0644))
+	t.Log("output written to testdata/maestro_filtered.json")
```
Comment (Skisocks) [inline pkg/swagfilter/filter_test.go]: Rather than overwriting the file here I'd just assert that the filtered file == the output of the strip command. Then you don't need to do the assertions below and it keeps it to one source of truth for the test.

**Review by Skisocks — CHANGES_REQUESTED** [review]

**Review by Skisocks — APPROVED** [review]
